test/unit/role_test.rb

```diff
@@ -126,7 +126,7 @@ class RoleTest < ActiveSupport::TestCase
   end
 
   test "should not be destroyable when it has appointments" do
-    role = create(:role, role_appointments: [create(:role_appointment)])
+    role = create(:role, :occupied)
```

Comment:
Is this functionally the same? It's a little unclear to me whether this fixes an error or just something where we thought an alternative syntax would be neater?

Reply:
I think this relates to the issue you spotted a few days ago: `:occupied` specifically adds a `role_appointment` in a hook (and sets the dates appropriately), so at the very least it's a neater way of writing the same thing; but it was also leading to a validation error that looked the same as elsewhere and needed adjusting the hooks, so I think it's just incorrect in tests to create a role in the original way.